File: RestaurantHubBackend/src/main/java/com/restauranthub/service/StaffService.java
package com.restauranthub.service;

import java.util.List;
import java.util.Optional;

import com.restauranthub.model.Staff;

public interface StaffService {

    Staff addStaff(Staff staff);

    List<Staff> getAllStaff();

    Optional<Staff> getStaffById(Long id);

    Staff updateStaff(Long id, Staff staff);

    String deleteStaff(Long id);

}

File: frontend/src/services/staffService.js
import API from "./api";

export const getStaff = () => {

    return API.get("/staff");

};

export const addStaff = (data) => {

    return API.post("/staff", data);

};

export const updateStaff = (id, data) => {

    return API.put(`/staff/${id}`, data);

};

export const deleteStaff = (id) => {

    return API.delete(`/staff/${id}`);

};